recipes/doctest/2.3.5/conanfile.py

```diff
@@ -0,0 +1,67 @@
+from conans import ConanFile, tools, CMake
+import os
+
+
+class DoctestConan(ConanFile):
+    name = "doctest"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/onqtam/doctest"
+    description = "C++11/14/17/20 single header testing framework"
+    license = "MIT"
+    settings = ("os", "compiler", "build_type", "arch")
+    generators = "cmake"
+    options = {"main_in_static_lib": [True, False]}
+    default_options = {"main_in_static_lib": False}
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+
+    @property
+    def is_mingw(self):
+        return self.settings.os == "Windows" and self.settings.compiler == "gcc"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["DOCTEST_WITH_TESTS"] = False
+        cmake.definitions["DOCTEST_WITH_MAIN_IN_STATIC_LIB"] = self.options.main_in_static_lib
+        # doctest installs CMake config files, which are disallowed by conan-center
+        # install in a temporary folder, and package after
+        cmake.definitions["CMAKE_INSTALL_PREFIX"] = "install"
+        cmake.configure(build_folder=self._build_subfolder, source_folder=self._source_subfolder)
+        return cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+        # the CMake target has EXCLUDE_FROM_ALL, must call it explicitly
+        if self.options.main_in_static_lib:
+            cmake.build(target="doctest_with_main")
```

Comment:
do we need to call both `cmake.build()` and `cmake.build(target="doctest_with_main")` or second one alone is sufficient?

Reply:
Yes, since the single-header assembling is a custom target that we have to "build"

Reply:
Yes, since the single-header assembling is a custom target that we have to "build"